File: src/main/java/org/pcha/benthos/language/BloblangFileType.java
package org.pcha.benthos.language;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class BloblangFileType extends LanguageFileType {
    public static final BloblangFileType INSTANCE = new BloblangFileType();

    private BloblangFileType() {
        super(BloblangLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Bloblang";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Bloblang mapping file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "blobl";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return BloblangIcons.FILE;
    }
}

File: src/main/kotlin/com/github/pcha/intellij/bloblang/BloblangFileType.kt
package com.github.pcha.intellij.bloblang

import com.intellij.openapi.fileTypes.LanguageFileType
import javax.swing.Icon

class BloblangFileType private constructor() : LanguageFileType(BloblangLanguage.INSTANCE) {
    companion object {
        val INSTANCE = BloblangFileType()
    }

    override fun getName(): String {
        return "Bloblang File"
    }

    override fun getDescription(): String {
        return "Bloblang file"
    }

    override fun getDefaultExtension(): String {
        return "blobl"
    }

    override fun getIcon(): Icon {
        return BloblangIcons.Icon
    }
}